Bottom: b22cf82204edb7f072cb9aef7af802da567e0d4c
Top:    c88643d73161fd843c0a773f44ce588ba0864d59
Author: Peter Clifton <pcjc2@cam.ac.uk>
Date:   2011-08-08 23:06:44 +0100

Refresh of tmp

---

diff --git a/src/hid/gtk/gtkhid-main.c b/src/hid/gtk/gtkhid-main.c
index ed28ca1..9e312e6 100644
--- a/src/hid/gtk/gtkhid-main.c
+++ b/src/hid/gtk/gtkhid-main.c
@@ -332,11 +332,7 @@ ghid_set_crosshair (int x, int y, int action)
       widget_y = pointer_y - offset_y;
 
       ghid_event_to_pcb_coords (widget_x, widget_y, &pcb_x, &pcb_y);
-
-      gport->view_x0 = MAX (0, SIDE_X (pcb_x) - widget_x * gport->zoom);
-      gport->view_y0 = MAX (0, SIDE_Y (pcb_y) - widget_y * gport->zoom);
-
-      ghid_pan_fixup ();
+      ghid_pan_view_abs (pcb_x, pcb_y, widget_x, widget_y);
 
       /* Just in case we couldn't pan the board the whole way,
        * we warp the pointer to where the crosshair DID land.
@@ -1599,7 +1595,7 @@ currently within the window already.
 %end-doc */
 
 static int
-Center(int argc, char **argv, int pcb_x, int pcb_y)
+Center(int argc, char **argv, int x, int y)
 {
   GdkDisplay *display;
   GdkScreen *screen;
@@ -1614,10 +1610,7 @@ Center(int argc, char **argv, int pcb_x, int pcb_y)
   widget_x = gport->width / 2;
   widget_y = gport->height / 2;
 
-  gport->view_x0 = MAX (0, SIDE_X (pcb_x) - widget_x * gport->zoom);
-  gport->view_y0 = MAX (0, SIDE_Y (pcb_y) - widget_y * gport->zoom);
-
-  ghid_pan_fixup ();
+  ghid_pan_view_abs (x, y, widget_x, widget_y);
 
   /* Now move the mouse pointer to the place where the board location
    * actually ended up.
@@ -1625,7 +1618,7 @@ Center(int argc, char **argv, int pcb_x, int pcb_y)
    * XXX: Should only do this if we confirm we are inside our window?
    */
 
-  ghid_pcb_to_event_coords (pcb_x, pcb_y, &widget_x, &widget_y);
+  ghid_pcb_to_event_coords (x, y, &widget_x, &widget_y);
   gdk_window_get_origin (gport->drawing_area->window, &offset_x, &offset_y);
 
   pointer_x = offset_x + widget_x;
